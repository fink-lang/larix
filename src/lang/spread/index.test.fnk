{describe, it, expect, to_match_snapshot} = import '@fink/jest'

{parse_expr, parse} = import '../../'


describe:: 'spread', fn:

  it:: 'parses single line: ...foobar', fn:
    expect::
      parse_expr:: `...foobar`
      to_match_snapshot


  it:: 'parses standalone', fn:
    expect::
      parse_expr:: `[a, ..., b]`
      to_match_snapshot


  it:: 'parses multiple', fn:
    expect::
      parse_expr:: `
        (
          ...a
          ...b
        )`
      to_match_snapshot

