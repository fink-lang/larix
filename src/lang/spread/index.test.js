import {parse_expr} from '../../';


describe('spread', ()=> {

  it('parses single line: ...foobar', ()=> {
    expect(
      parse_expr(`...foobar`)
    ).toMatchSnapshot();
  });
});
