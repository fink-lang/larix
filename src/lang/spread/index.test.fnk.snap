// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spread parses multiple 1`] = `
group (1:0-4:1)
  spread ... (2:2-2:6)
    ident (2:5-2:6) a
  spread ... (3:2-3:6)
    ident (3:5-3:6) b
`;

exports[`spread parses single line: ...foobar 1`] = `
spread ... (1:0-1:9)
  ident (1:3-1:9) foobar
`;

exports[`spread parses standalone 1`] = `
array (1:0-1:11)
  ident (1:1-1:2) a
  spread ... (1:4-1:7)
  ident (1:9-1:10) b
`;
